src/components/map/Map.tsx

```diff
@@ -157,138 +107,219 @@ export const Map = ({
+          locationSettings.currentLocation.longitude
+        ],
+        zoomLevel: 15,
+        animationDuration: 1000
+      });
+    }
+  };
+
   return (
     <View style={[styles.container, style]}>
-      <MapView
-        clusterColor={colors.primary}
-        clusterFontFamily="regular"
-        clusteringEnabled={clusteringEnabled}
-        radius={clusterDistance}
-        renderCluster={renderCluster}
-        initialRegion={initialRegion}
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
+      <LoadingSpinner loading={isLoading} />
+      <MapLibreGL.MapView 
+        style={styles.map}
+        styleJSON='https://tileserver-gl.smart-village.app/styles/osm-liberty/style.json'
         onPress={onMapPress}
         ref={refForMapView}
-        rotateEnabled={false}
-        showsBuildings={false}
-        showsPointsOfInterest={false}
-        showsUserLocation={showsUserLocation}
-        showsMyLocationButton={false}
-        toolbarEnabled={false}
-        style={[stylesForMap().map, mapStyle]}
-        userLocationPriority="balanced"
-        minZoom={minZoom}
-        mapPadding={{
-          top: 0,
-          right: 0,
-          bottom: -normalize(30),
-          left: 0
-        }}
-        legalLabelInsets={{
-          top: 0,
-          right: 0,
-          bottom: normalize(10),
-          left: 0
-        }}
+        onRegionDidChange={(region) => {
+          const newZoomLevel = Math.round(region.properties.zoomLevel);
+          if (!isNaN(newZoomLevel)) {
+            setZoomLevel(newZoomLevel);
+          }
+        }}  
+        showUserLocation={showsUserLocationSetting}
       >
-        <UrlTile
-          urlTemplate="https://tileserver-gl.smart-village.app/styles/osm-liberty/{z}/{x}/{y}.png"
-          shouldReplaceMapContent={device.platform === 'ios'}
+        <MapLibreGL.Camera
+          ref={cameraRef}
         />
-        {!!geometryTourData?.length && (
-          <Polyline
-            coordinates={geometryTourData}
-            strokeWidth={2}
-            strokeColor={colors.primary}
-            zIndex={1}
-          />
-        )}
-        {locations?.map((marker, index) => {
+        {clusters.map((cluster, index) => {
+          const [longitude, latitude] = cluster.geometry.coordinates;
+          const { cluster: isCluster, point_count: pointCount } = cluster.properties;
+
+          if (isCluster) {
+            return (
+              <MapLibreGL.PointAnnotation
+                key={`cluster-${index}`}
+                id={`cluster-${index}`}
+                coordinate={[longitude, latitude]}
+              >
+                <View style={styles.clusterContainer}>
+                  <View style={styles.cluster}>
+                    <Text style={styles.clusterText}>{pointCount}</Text>
+                  </View>
+                </View>
+              </MapLibreGL.PointAnnotation>
+            );
+          }
```

Comment:
Is the new cluster structure we are using here fully compatible with the previous cluster? As far as I can see from the screenshots, the design is not exactly the same.

Reply:
yes, implemented all functionalities and adjusted the design in the newest update https://github.com/smart-village-solutions/smart-village-app-app/pull/1154/commits/7d15a3d5f2af482f630518fdc922da1732420e58